.github/workflows/ci.yml

```diff
@@ -34,7 +34,7 @@ jobs:
   # less overhead to keep things in GH actions. See work on building these
   # in GCP with Cloud Build: https://github.com/firezone/firezone/pull/2234
   build-images:
-    runs-on: ubuntu-22.04-firezone-4c
```

Comment:
Can we keep just this one for now and see how much it will cost us? This is a task that benefits the most from having more cores.

Reply:
Hmm it seems Docker image caching is broken for the rust builds:

I'll take a look at why that is before merging this.

@AndrewDryga The issue is that our build-images are mostly I/O (network, memory, or disk) bound don't scale well by throwing cores at the problem.

Here's a worst-case example on the paid 4c runner since the cache is not used:

https://github.com/firezone/firezone/actions/runs/6714739908/job/18248404304?pr=2544

9 min, 1 sec

And with the free tier runner:

https://github.com/firezone/firezone/actions/runs/6738371788/job/18317802417#step:6:1

11 min, 31 sec

So we're paying $500-600 / month to save ~ 20% in CI time in the best case scenario.

The difference is much less when the caches hit.

Again, not saying we shouldn't spend $ to make CI faster, just that the value here is particularly lacking. 

For $500 / month we can have 4 dedicated bare metal servers on Vultr provisioned via Terraform that will run circles around these runners: https://www.vultr.com/products/bare-metal/#pricing

Reply:
Besides, the kotlin and swift builds are the bottleneck, so even if build-images and elixir build times were `0s`, CI time wouldn't improve. In light of that, going to go ahead and merge this.